components/nodemanager-service/pgdb/nodes.go

```diff
@@ -153,22 +153,22 @@ var nodesSortFields = map[string]string{
 }
 
 type dbNode struct {
-	ConnectionError string          `db:"connection_error"`
-	ID              string          `db:"id"`
-	LastContact     time.Time       `db:"last_contact"`
-	LastJob         string          `db:"last_job"`
-	Manager         string          `db:"manager"`
-	ManagerIDs      json.RawMessage `db:"manager_ids"`
-	Name            string          `db:"name"`
-	Platform        string          `db:"platform"`
-	PlatformVersion string          `db:"platform_version"`
-	Projects        json.RawMessage `db:"projects"`
-	ReportID        string          `db:"report_id"`
-	State           string          `db:"source_state"`
-	Status          string          `db:"status"`
-	Tags            json.RawMessage `db:"tags"`
-	TargetConfig    json.RawMessage `db:"target_config"`
-	TotalCount      int64           `db:"total_count"`
+	ConnectionError string           `db:"connection_error"`
+	ID              string           `db:"id"`
+	LastContact     time.Time        `db:"last_contact"`
+	LastJob         string           `db:"last_job"`
+	Manager         string           `db:"manager"`
+	ManagerIDs      json.RawMessage  `db:"manager_ids"`
+	Name            string           `db:"name"`
+	Platform        string           `db:"platform"`
+	PlatformVersion string           `db:"platform_version"`
+	Projects        json.RawMessage  `db:"projects"`
+	ReportID        string           `db:"report_id"`
+	State           string           `db:"source_state"`
+	Status          string           `db:"status"`
+	Tags            json.RawMessage  `db:"tags"`
+	TargetConfig    *json.RawMessage `db:"target_config"`
```

Comment:
Could this change be avoided by checking for `if len(inNode.TargetConfig) > 0 {`?

I guess the deeper question is something like: do we need to differentiate between an empty TargetConfig vs nil?

Reply:
we do - because without doing the pointer there you end up with a nasty sql error about scanning a nil value. it's because we didnt give it a default value when we first set this all up.